rust-runtime/aws-smithy-types-convert/src/date_time.rs

```diff
@@ -12,25 +12,33 @@ use aws_smithy_types::DateTime;
 use std::error::Error as StdError;
 use std::fmt;
 
-/// Conversion error
-#[non_exhaustive]
 #[derive(Debug)]
-pub enum Error {
+enum ErrorKind {
     /// Conversion failed because the value being converted is out of range for its destination
-    #[non_exhaustive]
     OutOfRange(Box<dyn StdError + Send + Sync + 'static>),
 }
 
-impl StdError for Error {}
+/// Conversion error
+#[derive(Debug)]
+pub struct Error {
```

Comment:
I am perfectly fine with this to be here, but the RFC says to let you know: "Errors and their context structs are in an error submodule for any given module. They are not mixed with other non-error code"